src/lib/actions/FocusTrap/focusTrap.ts

```diff
@@ -0,0 +1,45 @@
+// Action: Focus Trap
+
+export function focusTrap(node: HTMLElement, enabled: boolean) {
+	const elemWhitelist: string = 'a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"])';
+	let elemFirst: HTMLElement;
+	let elemLast: HTMLElement;
+
+	function lastLastKeydown(e: KeyboardEvent): void {
+		if (e.code === 'Tab') {
+			e.preventDefault();
+			elemFirst.focus();
+		}
+	}
+
+	const onInit = () => {
+		if (enabled === false) return;
+		// Gather all focusable elements
+		const focusableElems: any = node.querySelectorAll(elemWhitelist);
```

Comment:
Since you're using a querySelectorAll, this would be a `NodeList`

Reply:
Immediately causes line 21/22 to error. Doesn't like those are set as `HTMLElement`, it suggests `Node`. But that change causes a bunch of stuff to error. So a bit of a rabbit hole. I might need you to review directly.

Reply:
Yep, NodeLists are different to arrays- `const focusableElems: HTMLElement[] = Array.from(node.querySelectorAll(elemWhitelist);)` would get the job done.